pymc3/step_methods/hmc/trajectory.py

```diff
@@ -57,6 +57,13 @@ def _theano_energy_function(H, q, **theano_kwargs):
     return energy_function, p
 
 
+def _theano_velocity_function(H, p, **theano_kwargs):
+    v = H.pot.velocity(p)
```

Comment:
it looks like `H.pot.velocity` is already a theano function, but it also looks like you're using the velocity calculation separately -- is it necessary to have it attached to the named tuple `H` as well as calculated separately?